src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs

```diff
@@ -88,12 +88,17 @@ private BlobHandle GetOrAddBlob(ImmutableArray<byte> blob)
                 index = MetadataTokens.BlobHandle(_blobHeapSize);
                 _blobs.Add(blob, index);
 
-                _blobHeapSize += (blob.Length <= 0x7f) ? 1 : ((blob.Length <= 0x3fff) ? 2 : 4) + blob.Length;
```

Comment:
I stared at this line for a long time yesterday without figuring out that it had an issue. Reading the correction, I'm still not seeing the nature of the issue.